src/main/kotlin/org/elm/lang/core/psi/elements/ElmFieldAccessExpr.kt

```diff
@@ -2,49 +2,22 @@ package org.elm.lang.core.psi.elements
 
 import com.intellij.lang.ASTNode
 import com.intellij.psi.PsiElement
-import org.elm.lang.core.psi.ElmFunctionCallTargetTag
-import org.elm.lang.core.psi.ElmNamedElement
 import org.elm.lang.core.psi.ElmAtomTag
+import org.elm.lang.core.psi.ElmFieldAccessPartTag
+import org.elm.lang.core.psi.ElmFunctionCallTargetTag
 import org.elm.lang.core.psi.ElmPsiElementImpl
 import org.elm.lang.core.psi.ElmTypes.LOWER_CASE_IDENTIFIER
-import org.elm.lang.core.resolve.ElmReferenceElement
-import org.elm.lang.core.resolve.reference.ElmReferenceCached
-import org.elm.lang.core.resolve.scope.ExpressionScope
-
 
 /**
  * Accessing one or more fields on a base record.
  *
  * e.g. `model.currentUser.name`
  */
-class ElmFieldAccessExpr(node: ASTNode) : ElmPsiElementImpl(node), ElmReferenceElement, ElmAtomTag, ElmFunctionCallTargetTag {
-
-    val lowerCaseIdentifierList: List<PsiElement>
-        get() = findChildrenByType(LOWER_CASE_IDENTIFIER)
-
-
-    override val referenceNameElement: PsiElement
-        get() = lowerCaseIdentifierList.first()
-
-    override val referenceName: String
-        get() = referenceNameElement.text
-
-    override fun getReference() =
-            BaseRecordReference(this)
-}
-
-
-class BaseRecordReference(element: ElmReferenceElement)
-    : ElmReferenceCached<ElmReferenceElement>(element) {
+class ElmFieldAccessExpr(node: ASTNode) : ElmPsiElementImpl(node), ElmAtomTag, ElmFunctionCallTargetTag, ElmFieldAccessPartTag {
 
-    override fun resolveInner(): ElmNamedElement? {
-        return getVariants().find { it.name == element.referenceName }
-    }
+    val leftExpr: ElmFieldAccessPartTag
```

Comment:
If I understand this correctly, this is `model.currentUser` in `model.currentUser.name`? Maybe naming it `target` or `targetExpr` to match `ElmFunctionCall` would be more clear.

Reply:
👍